code/datums/traits/_quirk.dm

```diff
@@ -65,22 +65,78 @@
 		return
 	on_process()
 
-/mob/living/proc/get_trait_string(medical) //helper string. gets a string of all the traits the mob has
+/mob/living/proc/get_trait_string(medical, category) //helper string. gets a string of all the traits the mob has
 	var/list/dat = list()
-	if(!medical)
-		for(var/V in roundstart_quirks)
-			var/datum/quirk/T = V
-			dat += T.name
-		if(!dat.len)
-			return "None"
-		return dat.Join(", ")
-	else
-		for(var/V in roundstart_quirks)
-			var/datum/quirk/T = V
-			dat += T.medical_record_text
-		if(!dat.len)
-			return "None"
-		return dat.Join("<br>")
+	switch(category)
+		if(0)
+			if(!medical)
+				for(var/V in roundstart_quirks)
+					var/datum/quirk/T = V
+					dat += T.name
+				if(!dat.len)
+					return "None"
+				return dat.Join(", ")
+			else
+				for(var/V in roundstart_quirks)
+					var/datum/quirk/T = V
+					dat += T.medical_record_text
+				if(!dat.len)
+					return "None"
+				return dat.Join("<br>")
+		//Major Disabilities
+		if(3)
+			if(!medical)
+				for(var/V in roundstart_quirks)
+					var/datum/quirk/T = V
+					if(T.value < -1)
+						dat += T.name
+				if(!dat.len)
+					return "None"
+				return dat.Join(", ")
+			else
+				for(var/V in roundstart_quirks)
+					var/datum/quirk/T = V
+					if(T.value < -1)
+						dat += T.medical_record_text
+					if(!dat.len)
+						return "No major disabilities have been declared."
+				return dat.Join("<br>")
+		//Minor Disabilities
+		if(2)
+			if(!medical)
+				for(var/V in roundstart_quirks)
+					var/datum/quirk/T = V
+					if(T.value == -1)
+						dat += T.name
+					if(!dat.len)
+						return "None"
+				return dat.Join(",")
+			else
+				for(var/V in roundstart_quirks)
+					var/datum/quirk/T = V
+					if(T.value == -1)
+						dat += T.medical_record_text
+					if(!dat.len)
+						return "No minor disabilities have been declared."
+				return dat.Join("<br>")
+		//Neutral and Positive quirks
+		if(1)
+			if(!medical)
+				for(var/V in roundstart_quirks)
+					var/datum/quirk/T = V
+					if(T.value > -1)
+						dat += T.name
+					if(!dat.len)
+						return "None"
+				return dat.Join(", ")
+			else
+				for(var/V in roundstart_quirks)
+					var/datum/quirk/T = V
+					if(T.value > -1)
+						dat += T.medical_record_text
```

Comment:
Your way is better performant, but I'd use ternary operators for higher clarity and less duplicate code.
Examples:
`dat += medical ? T.medical_record_text : T.name`
`return medical ? "No minor disabilities have been declared." : "None"`

Reply:
Thank you!